### ISSUE TITLE
Potential false positives for `expect-expect` and `valid-title` when using `it.scoped`

### ISSUE THREAD

**Issue by dmurtari** [vitest-dev/eslint-plugin-vitest#821]
When using `it.scoped`, I'm getting linting errors for both `vitest/expect-expect` and `vitest/valid-title`. This issue first appeared when we upgraded to v1.4.0.

I'm able to reproduce using the "Scoping Values to Suite" [example from the Vitest docs](https://vitest.dev/guide/test-context.html#scoping-values-to-suite), with the `configs.recommended` ruleset:

```ts
describe('use scoped values', () => {
  test.scoped({ dependency: 'new' }) // <-- this line has linting errors

  test('uses scoped value', ({ dependant }) => {
    // `dependant` uses the new overridden value that is scoped
    // to all tests in this suite
    expect(dependant).toEqual({ dependency: 'new' })
  })
})
```

```ts
export default [
  {
    files: ["tests/**/*.ts"],
    plugins: {
      vitest,
    },
    rules: {
      ...vitest.configs.recommended.rules,
    },
  },
];
```

<details>
  <summary>Screenshot from my editor</summary>
  <img width="1015" height="656" alt="Image" src="https://github.com/user-attachments/assets/8f881f52-37cc-4b6a-987d-e8e15b97bb56" />
</details>

Happy to take a look at this as well, if this is actually a false positive!
